File: NSNotification/BlockExample/Example/ViewController.m
//
//  ViewController.m
//  Example
//
//  Created by ShannonChen on 2018/2/1.
//  Copyright © 2018年 ShannonChen. All rights reserved.
//

#import "ViewController.h"
#import "DetailViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
}

- (IBAction)pushDetailViewController:(id)sender {
    
    [self.navigationController pushViewController:[DetailViewController new] animated:YES];
}


@end

File: CoreAnimationAdvancedTechniques/09_Transition/ImageTransition/ImageTransition/ViewController.swift
//
//  ViewController.swift
//  ImageTransition
//
//  Created by ShannonChen on 2017/1/29.
//  Copyright © 2017年 ShannonChen. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    // MARK: Properties
    let images = [#imageLiteral(resourceName: "Anchor"),
                  #imageLiteral(resourceName: "Cone"),
                  #imageLiteral(resourceName: "Igloo"),
                  #imageLiteral(resourceName: "Spaceship")]
    
    @IBOutlet weak var imageView: UIImageView!

    override func viewDidLoad() {
        super.viewDidLoad()

        
    }

    @IBAction func switchImage(_ sender: UIButton) {
        
        // setup crossfade transition 
        let transition = CATransition()
        transition.type = kCATransitionFade
        
        // apply transition to image view backing layer
        imageView.layer.add(transition, forKey: nil)
        
        // cycle to next image
        
        var nextIndex = 0
        if let currentImage = imageView.image, let currentIndex = images.index(of: currentImage) {
            nextIndex = (currentIndex + 1) % images.count
        }
        
        imageView.image = images[nextIndex]
        
    }

}

